multicluster/controllers/multicluster/resourceexport_controller.go

```diff
@@ -71,20 +71,29 @@ func NewResourceExportReconciler(
 
 // Reconcile is part of the main kubernetes reconciliation loop which aims to
 // move the current state of the cluster closer to the desired state.
-// TODO(user): Modify the Reconcile function to compare the state specified by
-// the ResourceExport object against the actual cluster state, and then
-// perform operations to make the cluster state reflect the state specified by
-// the user.
-//
-// For more details, check Reconcile and its Result here:
-// - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.8.3/pkg/reconcile
+// Reconcile will process all kinds of ResourceExport. Service and Endpoint kinds of ResourceExport
+// will be handled in this file, and all other kinds will have their own handler files, eg: newkind_handler.go
 func (r *ResourceExportReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
 	klog.V(2).InfoS("reconciling ResourceExport", "resourceexport", req.NamespacedName)
 	var resExport mcsv1alpha1.ResourceExport
 	if err := r.Client.Get(ctx, req.NamespacedName, &resExport); err != nil {
 		return ctrl.Result{}, client.IgnoreNotFound(err)
 	}
 
+	switch resExport.Spec.Kind {
+	case common.ServiceKind:
+		klog.V(2).InfoS("Reconciling Service type of ResourceExport", "resourceexport", req.NamespacedName)
+	case common.EndpointsKind:
+		klog.V(2).InfoS("Reconciling Endpoint type of ResourceExport", "resourceexport", req.NamespacedName)
+	// Developer can add more supported kinds here in the future.
+	// eg: add a new case and a new method 'handleNewKind' in a new file like 'newkind_handler.go'
+	//  case common.NewKind:
+	//    return r.handleNewKind(ctx, req, resExport, err)
+	default:
+		klog.V(2).InfoS("It's not expected kind, skip reconciling ResourceExport", "resourceexport", req.NamespacedName)
```

Comment:
Should this unexpected event be logged with Error at default level to get attention?

Reply:
done, change it to default level.